### PR TITLE
Fix model ref of existing models

### PR DESCRIPTION
**Pull request by CGNonofr**

### CODE REVIEW

File: src/service-override/modelEditor.ts [hunk 1]
```diff
@@ -150,7 +155,18 @@ class EditorService implements IEditorService {
  * This class just force dispose the refs when the model is disposed
  */
 class CustomTextModelResolverService extends TextModelResolverService {
-  override async createModelReference (resource: URI) {
+  constructor (private _instantiationService: IInstantiationService, fileService: IFileService, undoRedoService: IUndoRedoService, private _modelService: IModelService, uriIdentityService: IUriIdentityService) {
+    super(_instantiationService, fileService, undoRedoService, _modelService, uriIdentityService)
+  }
+
+  override async createModelReference (resource: URI): Promise<IReference<IResolvedTextEditorModel>> {
+    // If the model already exists, return an immortable reference to it
```
Comment (samuel-olivier) [inline src/service-override/modelEditor.ts:172]: immortable ?

File: src/service-override/modelEditor.ts [hunk 2]
```diff
@@ -150,7 +155,18 @@ class EditorService implements IEditorService {
...
+    const existingModel = this._modelService.getModel(resource)
+    if (existingModel != null) {
+      return new ImmortalReference(this._instantiationService.createInstance(TextResourceEditorModel, resource) as IResolvedTextEditorModel)
+    }
+
+    // Otherwise, try to create a referene to the model
```
Comment (samuel-olivier) [inline src/service-override/modelEditor.ts:178]: referene ?

File: src/service-override/modelEditor.ts [hunk 3]
```diff
@@ -119,6 +125,8 @@ class EditorService implements IEditorService {
       return undefined
     }
 
+    // Otherwise, let the user destroy the model, never destry the reference
```
Comment (samuel-olivier) [inline src/service-override/modelEditor.ts:128]: destry ?

File: src/service-override/modelEditor.ts (hunk 1 above)
Comment (CGNonofr) [reply src/service-override/modelEditor.ts]: d95ddc74ad6f2c48a6dceba545e45e06bfbe6fd7

File: src/service-override/modelEditor.ts (hunk 2 above)
Comment (CGNonofr) [reply src/service-override/modelEditor.ts]: d95ddc74ad6f2c48a6dceba545e45e06bfbe6fd7

File: src/service-override/modelEditor.ts (hunk 3 above)
Comment (CGNonofr) [reply src/service-override/modelEditor.ts]: d95ddc74ad6f2c48a6dceba545e45e06bfbe6fd7

**Review by samuel-olivier — APPROVED** [review]

**Pull request merged**